src/ai_commands.js

```diff
@@ -377,74 +377,111 @@ const AiOut = ({ id, aiOutProcess }) => {
+    //generate ID, in some rare circumstances previous ID could still be present, try remove it
+    id = (prefValue ? prefValue + " " : "") + "-profileDirID." + new Date().getTime() + performance.now(),
+    dir = (process.env.VSCODE_PORTABLE ? process.env.VSCODE_PORTABLE + "/user-data" : process.env.APPDATA + "/Code") + "/User/",
+    settingsJsonWatcher = workspace.createFileSystemWatcher(new RelativePattern(dir, "**/settings.json")),
+    settingsJsonWatcherEventHandler = uri => {
+      if (profileDir)
+        return;
+
+      fs.readFile(uri.fsPath, (err, data) => {
+        if (profileDir)
+          return;
+
+        const json = parse(data.toString());
+        if (json[pref.key] !== id )
+          return;
+
+        profileDir = uri.fsPath.replace(/[^\\/]+$/, "");
+        settingsJsonWatcher.dispose();
+        reset();
+      });
+    },
+    reset = () => {
+      clearTimeout(settingsTimer);
+      prefs.update(prefName, prefValue, true);
+      initKeybindings(profileDir || dir);
+    };
+
+  settingsJsonWatcher.onDidChange(settingsJsonWatcherEventHandler);
+  settingsJsonWatcher.onDidCreate(settingsJsonWatcherEventHandler);
+  prefs.update(prefName, id, true);
+  const settingsTimer = setTimeout(reset, 2000);
+  const initKeybindings = dir => {
+    let readFileLast = 0; //prevent multiple calls
+    const commandsList = {};
+    for(let i = 0; i < _commandsList.length; i++)
+      commandsList[commandsPrefix + _commandsList[i]] = "";
+  
+    const keybindingsDefaultRaw = require("../package.json").contributes.keybindings;
+    const keybindingsDefault = keybindingsDefaultRaw.reduce((a,b)=>(a[b.command]=b.key,a),{});
+    const fs = require("fs");
+    const promise = {resolve: ()=>{},isResolved:false};
+  
+    const readFile = uri => {
+  
+      const now = performance.now();
+      if (uri && (uri.scheme != "file" || uri.fsPath != file || !promise.isResolved || readFileLast + 200 > now))
+        return;
+  
+      keybindings = new Promise(resolve => (promise.resolve = resolve, promise.isResolved = false));
+      Object.assign(keybindings, Object.assign({}, keybindingsDefault));
+      readFileLast = now;
+      //read file
+      fs.readFile(file, (err, data) => {
+        //we can't use JSON.parse() because file may contain comments
+        keybindingsUpdate(err ? keybindingsDefaultRaw : parse(data.toString()));
+      });
+    };
+    const fileName = "keybindings.json";
+    const file = Uri.file(dir + fileName).fsPath;
+    const watcher = workspace.createFileSystemWatcher(new RelativePattern(dir, "*.json"));
+    watcher.onDidChange(readFile);
+    watcher.onDidCreate(readFile);
+    watcher.onDidDelete(readFile);
+
+    const keybindingsUpdate = list => {
+      const keybindingsNew = {},
+        keybindingsFallback = Object.assign({}, keybindingsDefault);
+
+      for(let i = 0; i < list.length; i++) {
```

Comment:
@vanowm I'm getting the error `Exception has occurred: TypeError: Cannot read properties of undefined (reading 'length')` on this line. Looks to me like `list` is null.

Reply:
Yes, on startup. Here's the output you requested --

```
{file: 'c:\Users\danpo\AppData\Roaming\Code\User\\keybindings.json', err: null, data: Buffer(65), toString: '// Place your key bindings in this file to override the defaults
'}
```

Reply:
and just to make sure, the keybindings.json is a valid json w/comments?

Reply:
The file referenced above is empty other than the line `// Place your key bindings in this file to override the defaults`